.github/workflows/ci.yml

```diff
@@ -85,3 +85,26 @@ jobs:
 
       - name: Run a single unit test to verify the testing setup
         run: ./vendor/bin/phpunit -c ./web/core ./web/core/modules/user/tests/src/Unit/UserAccessControlHandlerTest.php
+
+  # Using outdated Composer version to test the Composer version constraint.
+  test-composer:
```

Comment:
there is no simple way to use existing jobs without adding a lot of `if` to every step as this check happens early in the run.

instead, having a standalone (fast) job is easier to maintain.

---

I called this `test-composer` rather than `test-composer-fail` etc. in case we want to add more composer-related tests in the future.